### PR TITLE
Plugin for integrating SwaggerHub with WSO2 API management cloud

### PR DESCRIPTION
**Pull request by Akila94**
> User has to have a WSO2 API management cloud account prior to integration.

> To enable an integration, a user must have a valid access token with apim:api_view and apim:api_create scopes.

> After enabling the integration in SwaggerHub, the user will be able to export the API to the cloud by executing the plugin in SwaggerHub.

### CODE REVIEW

File: swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+     * @param payload                   Body of the request
+     * @throws PluginExecutionException Thrown when an exception is caught while the plugin executes
+     */
+    public HttpResponse makePutRequest(String url, String tokenPrefix, String token, String contentType,
+                                       StringEntity payload) throws PluginExecutionException {
+        HttpResponse response;
+        HttpClient httpClient = HttpClients.createDefault();
+        HttpPut putRequest = new HttpPut(url);
+        putRequest.setHeader("Authorization", tokenPrefix + " " + token);
+        putRequest.setHeader("Content-Type", contentType);
```
Comment (dinusha92) [inline swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java]: Shall we move Authorization and Content-Type into constants?

File: swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+     * @return                          Returns the response of the request
+     * @throws PluginExecutionException Thrown when an exception is caught while the plugin executes
+     */
+    public HttpResponse makeGetRequest(String url, String tokenPrefix, String token, String contentType)
+            throws com.smartbear.swaggerhub.plugins.PluginExecutionException {
+        HttpResponse response;
+        HttpClient httpClient = HttpClients.createDefault();
+        HttpGet getRequest = new HttpGet(url);
+        getRequest.setHeader("Authorization", tokenPrefix + " " + token);
+        getRequest.setHeader("Content-Type", contentType);
```
Comment (dinusha92) [inline swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java]: keeping these as constants would be better.

File: swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/PayloadConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        structure.setTransport(schemes);
+        structure.setTiers(defaultTier);
+        structure.setVisibility(visibility);
+        structure.setEndpointConfig("");
+        CorsConfiguration configuration = new CorsConfiguration(false);
+        structure.setCorsConfiguration(configuration);
+
+        log.debug("Converting the POJO to a json string");
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
```
Comment (dinusha92) [inline swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/PayloadConfiguration.java]: If this conversion to JSON happens within try block, you should put the log within the try block. Shall we do so if that is the case?

File: swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java (hunk 2 above)
Comment (Akila94) [reply swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java]: Will do, thanks @dinusha92.

File: swaggerhub-wso2-integration/pom.xml [hunk 4]
```diff
@@ -0,0 +1,302 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+     <!--
+        /*
+        *  Copyright (c) 2005-2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (tgtshanika) [inline swaggerhub-wso2-integration/pom.xml]: Use correct license year

File: swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/PayloadConfiguration.java (hunk 3 above)
Comment (Akila94) [reply swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/PayloadConfiguration.java]: Yes, it should be in the try block, thanks.

File: swaggerhub-wso2-integration/pom.xml (hunk 4 above)
Comment (Akila94) [reply swaggerhub-wso2-integration/pom.xml]: Will correct it. Thanks @tgtshanika .

File: swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java (hunk 1 above)
Comment (Akila94) [reply swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java]: Moved

File: swaggerhub-wso2-integration/pom.xml (hunk 4 above)
Comment (Akila94) [reply swaggerhub-wso2-integration/pom.xml]: Corrected it.

**Pull request closed without merging**